### PR TITLE
Cleaning up code related to fix_pins and fix_clusters vpr options

### PR DESCRIPTION
**Pull request by sfkhalid**
Removed pad_loc_file string as it was no longer needed.

### CODE REVIEW

File: vpr/src/base/read_options.cpp [hunk 1]
```diff
@@ -2253,24 +2253,8 @@ void set_conditional_defaults(t_options& args) {
...
-        VPR_FATAL_ERROR(VPR_ERROR_UNKNOWN, "Unknown I/O pad location type\n");
-    }
-
     //Are the blocks locked to locations given by a constraints file?
-    if (std::string(args.constraints_file) == "not_locked") {
-        args.block_loc_type.set(NOT_LOCKED, Provenance::INFERRED);
-    } else {
-        args.block_loc_type.set(LOCKED, Provenance::INFERRED);
+    if (std::string(args.constraints_file) != "") {
         VTR_ASSERT(!args.constraints_file.value().empty());
```
Comment (vaughnbetz) [inline vpr/src/base/read_options.cpp:2258]: This assert doesn't look useful.  You are checking if the string is not empty, and then assert that it is not empty (the .empty() call boils down to the same check as the != "" call.  Delete this code block (if and assert) as it no longer does anything.

File: vpr/src/base/vpr_types.h [hunk 2]
```diff
@@ -837,6 +837,11 @@ enum e_place_algorithm {
     PATH_TIMING_DRIVEN_PLACE
 };
 
+enum e_pad_loc_type {
+	FREE,
```
Comment (vaughnbetz) [inline vpr/src/base/vpr_types.h]: Give a short comment of FREE and RANDOM (I/Os can move and optimized in placement, and I/Os are randomly locked down (to emulate board-level constraints).

**Comment by vaughnbetz** [comment]
Sanitized test failure is the known graphics memory leak issue, so OK.
Travis CI: PR failure is the llvm fetch failure (only affects compilation warning check), so OK.
I'll merge this without the comment change as Sarah is going to work on some code refactoring here next, so I'd like to land this and defer the comment upgrade to that refactoring. Waiting for the nightly tests to pass.

**Comment by sfkhalid** [comment]
> Sanitized test failure is the known graphics memory leak issue, so OK.
> Travis CI: PR failure is the llvm fetch failure (only affects compilation warning check), so OK.
> I'll merge this without the comment change as Sarah is going to work on some code refactoring here next, so I'd like to land this and defer the comment upgrade to that refactoring. Waiting for the nightly tests to pass.

@vaughnbetz Vaughn, the nightly tests have passed now. Would you be able to merge this code?

**Pull request merged**